google-cloud-datastore/src/main/java/com/google/cloud/datastore/aggregation/AggregationBuilder.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.datastore.aggregation;
+
+public interface AggregationBuilder<A extends Aggregation> {
```

Comment:
most builders in the library are simply static classes - do we need this level of abstraction?

Reply:
Hi @kolea2 , It will be required while adding aggregation to a query.

`
Query<AggregationResult> aggregationQuery = Query.newAggregationQueryBuilder()
  .addAggregation(count().as("total"))
  .addAggregation(sum("cost").as("total_cost"))
  .over(query)
  .build();
`

In this case, both sum and count are returning the instances of AggregationBuilder, and aggregationQuery can use the builders without being aware of the concrete builder implementation.

If we don't do this, then we need to have `N `overloaded version of `addAggregation` method for `N` type of aggregation query we will be supporting in future.